backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java

```diff
@@ -197,8 +195,47 @@ public ApiUser getCurrentApiUserInContainedTransaction() {
     return getCurrentApiUser();
   }
 
+  private ApiUser getPatientApiUser() {
+    TestOrder order = _contextHolder.getLinkedOrder();
+    PatientLink patientLink = _contextHolder.getPatientLink();
+    if (patientLink == null || order == null) {
+      throw new UnidentifiedUserException();
+    }
+
+    String username = getPatientLinkEmail(patientLink);
+    Optional<ApiUser> found = _apiUserRepo.findByLoginEmail(username);
+
+    if (found.isPresent()) {
+      LOG.debug("Patient has logged in before: retrieving user record.");
+      ApiUser user = found.get();
+      user.updateLastSeen();
+      user = _apiUserRepo.save(user);
+      return user;
+    } else {
+      LOG.info("Initial login for patient: creating user record.");
+      ApiUser user = new ApiUser(username, order.getPatient().getNameInfo());
+      user.updateLastSeen();
+      user = _apiUserRepo.save(user);
+
+      LOG.info(
+          "Patient user with id={} self-created from link {}",
+          user.getInternalId(),
+          patientLink.getInternalId());
+      return user;
+    }
+  }
+
+  private String getPatientLinkEmail(PatientLink link) {
+    return link.getInternalId() + "-noreply@simplereport.gov";
```

Comment:
I thought we were using the patient ID? Would need to be the link ID if we were linking to it explicitly, but having the patient be semi-durable seems useful.

Reply:
Ah good catch! Corrected